internal/docker/docker.go

```diff
@@ -0,0 +1,252 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"html/template"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	containerTypes "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/konstructio/colony/internal/constants"
+	"github.com/konstructio/colony/internal/download"
+	"github.com/konstructio/colony/internal/logger"
+)
+
+var ErrK3sContainerNotFound = fmt.Errorf("colony k3s container not found")
+
+type Client struct {
+	cli *client.Client
+	log *logger.Logger
+}
+
+type ColonyTokens struct {
+	LoadBalancerIP        string
+	LoadBalancerInterface string
+}
+
+func New(logger *logger.Logger) (*Client, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return nil, fmt.Errorf("error creating docker client: %w", err)
+	}
+
+	return &Client{
+		cli: cli,
+		log: logger,
+	}, nil
+}
+
+func getColonyK3sContainer(ctx context.Context, c *Client) (*types.Container, error) {
+
+	containers, err := c.cli.ContainerList(ctx, containerTypes.ListOptions{All: true})
+	if err != nil {
+		return nil, fmt.Errorf("error listing containers on host: %w", err)
+	}
+
+	for _, container := range containers {
+		if len(container.Names) > 0 && container.Names[0] == "/"+constants.ColonyK3sContainerName {
+			return &container, nil
+		}
+	}
+	return nil, ErrK3sContainerNotFound
+}
```

Comment:
Curious why _just this function_ isn't part of the `Client` object:

```suggestion
func (c *Client) getColonyK3sContainer(ctx context.Context) (*types.Container, error) {

	containers, err := c.cli.ContainerList(ctx, containerTypes.ListOptions{All: true})
	if err != nil {
		return nil, fmt.Errorf("error listing containers on host: %w", err)
	}

	for _, container := range containers {
		if len(container.Names) > 0 && container.Names[0] == "/"+constants.ColonyK3sContainerName {
			return &container, nil
		}
	}
	return nil, ErrK3sContainerNotFound
}
```

It's a one-liner 😄